R/g.part1.R

```diff
@@ -416,42 +430,24 @@ g.part1 = function(datadir = c(), outputdir = c(), f0 = 1, f1 = c(),
-      } else if (params_rawdata[["chunksize"]] > 0.6 & Nmetrics2calc >= 3 & Nmetrics2calc < 6) { # if user wants to extract 3-5 metrics
-        params_rawdata[["chunksize"]] = 0.5 # put limit to chunksize, because when processing in parallel memory is more limited
-      } else if (params_rawdata[["chunksize"]] > 0.6 & Nmetrics2calc >= 6) { # if user wants to extract more than 5 metrics
-        params_rawdata[["chunksize"]] = 0.4 # put limit to chunksize, because when processing in parallel memory is more limited
-      }
-      if (length(params_general[["maxNcores"]]) == 0) params_general[["maxNcores"]] = Ncores
-      Ncores2use = min(c(Ncores - 1, params_general[["maxNcores"]], (f1 - f0) + 1))
-      if (Ncores2use > 1) {
-        cl <- parallel::makeCluster(Ncores2use) # not to overload your computer
-        doParallel::registerDoParallel(cl)
-      } else {
-        # Don't process in parallel if only one core
-        params_general[["do.parallel"]] = FALSE
-      }
-      
-    } else {
-      cat(paste0("\nparallel processing not possible because number of available cores (",Ncores,") < 4"))
-      params_general[["do.parallel"]] = FALSE
+    Nmetrics2calc = sum(unlist(params_metrics[c("do.anglex", "do.angley", "do.anglez",
+                                                "do.zcx", "do.zcy", "do.zcz",
+                                                "do.enmo", "do.lfenmo", "do.en", "do.mad", "do.enmoa",
+                                                "do.roll_med_acc_x", "do.roll_med_acc_y", "do.roll_med_acc_z",
+                                                "do.dev_roll_med_acc_x", "do.dev_roll_med_acc_y", "do.dev_roll_med_acc_z",
+                                                "do.bfen", "do.hfen", "do.hfenplus", "do.lfen",
+                                                "do.lfx", "do.lfy", "do.lfz", "do.hfx", "do.hfy", "do.hfz",
+                                                "do.bfx", "do.bfy", "do.bfz", "do.brondcounts")]))
+    if (Nmetrics2calc > 4) { #Only give warning when user wants more than 4 metrics.
+      warning(paste0("\nExtracting many metrics puts higher demands on memory. Please consider",
+                     " reducing the value for argument chunksize or setting do.parallel to FALSE"))
+    }
+    if (params_rawdata[["chunksize"]] > 0.6 & Nmetrics2calc >= 3 & Nmetrics2calc < 6) { # if user wants to extract 3-5 metrics
+      params_rawdata[["chunksize"]] = 0.5 # put limit to chunksize, because when processing in parallel memory is more limited
+    } else if (params_rawdata[["chunksize"]] > 0.6 & Nmetrics2calc >= 6) { # if user wants to extract more than 5 metrics
+      params_rawdata[["chunksize"]] = 0.4 # put limit to chunksize, because when processing in parallel memory is more limited
     }
+    
     cat(paste0('\n Busy processing ... see ', outputdir, outputfolder,'/meta/basic', ' for progress\n'))
     `%myinfix%` = ifelse(params_general[["do.parallel"]], foreach::`%dopar%`, foreach::`%do%`)
```

Comment:
Since I know there is some urgency to getting a CRAN submission ready you could ignore this, but it seems a bit odd that there are two methods for processing sequentially. This one is used if the user uses the (default) `do.parallel = TRUE` but the code decides not to do that (because of cores, number of files, whatever). But the below 'simple loop' is used if `do.parallel = FALSE`. Could you instead escape the if statement here, and start a new one which would honour the changes made to do.parallel?

Reply:
Good point. That's a nice simplification.